File: backend/src/main/java/com/it/finance/personal_finance_be/dto/PatrimonialFundInitDto.java
package com.it.finance.personal_finance_be.dto;

import org.springframework.beans.BeanUtils;

public class PatrimonialFundInitDto extends PatrimonialFundDto {

    public PatrimonialFundInitDto(){}

    public PatrimonialFundInitDto(PatrimonialFundDto pf, MovementDto mov){
        BeanUtils.copyProperties(pf, this);
        this.setInitialMovement(mov);
    }

    private MovementDto initialMovement;

    public MovementDto getInitialMovement() {
        return initialMovement;
    }

    public void setInitialMovement(MovementDto initialMovement) {
        this.initialMovement = initialMovement;
    }

}

File: frontend/src/dto/finance/PatrimonialFundInitDto.ts
import PatrimonialFundDto from "@/dto/finance/PatrimonialFundDto.ts";
import type MovementDto from "@/dto/finance/MovementDto.ts";

export default class PatrimonialFundInitDto extends PatrimonialFundDto{

    initialMovement: MovementDto | null;

    constructor() {
        super();
    }

}